src/game/AI/BaseAI/CreatureAI.cpp

```diff
@@ -41,6 +41,10 @@ CreatureAI::CreatureAI(Creature* creature, uint32 combatActions) :
     SetMeleeEnabled(!(m_creature->GetCreatureInfo()->ExtraFlags & CREATURE_EXTRA_FLAG_NO_MELEE));
     if (m_creature->IsNoAggroOnSight())
         SetReactState(REACT_DEFENSIVE);
+    if (m_creature->GetSettings().HasFlag(CreatureStaticFlags2::SPAWN_DEFENSIVE))
+        SetReactState(REACT_DEFENSIVE);
+    else if (m_creature->GetSettings().HasFlag(CreatureStaticFlags::IGNORE_COMBAT))
+        SetReactState(REACT_PASSIVE);
```

Comment:
Passive is invalid here, SetCanEnterCombat

Reply:
done